src/Soil/Biogeochemistry/Biogeochemistry.jl

```diff
@@ -137,6 +158,311 @@ function DETECTModelParameters(;
+	θ_l = soil_moisture(model.driver, p, Y, t, z)
+        θ_ant_roots = antecedent_soil_moisture_roots(model.driver, p, Y, t, z)
+        θ_ant_microbe = antecedent_soil_moisture_microbes(model.driver, p, Y, t, z)
+        T_ant_soil = antecedent_soil_temperature(model.driver, p, Y, t, z)
+
+	Cr = soil_rootC(model.driver, p, Y, t, z)
+	Csom = soil_SOMC(model.driver, p, Y, t, z)
+	Cmic = soil_microbeC(model.driver, p, Y, t, z)
+
+	@. p.DETECT.D = co2_diffusivity(
+                                        T_soil,
+                                        θ_w,
+                                        params,
+                                        )
+
+	@. p.DETECT.Sᵣ = root_source(
+                                     T_soil,
+                                     T_ant_soil,
+                                     θ_l,
+                                     θ_ant_roots,
+                                     Cr,
+                                     params,
+                                     ) 
+
+	@. p.DETECT.Sₘ = microbe_source(
+                                        T_soil,
+                                        T_ant_soil,
+                                        θ_l,
+                                        θ_ant_microbe,
+                                        Csom,
+                                        Cmic,
+                                        params,
+                                        )
+    end
+    return update_aux!
+end
+
+"""
+    AbstractSoilCO2BC{FT} <: ClimaLSM. AbstractBC{FT}
+
+An abstract type for co2-specific types of boundary conditions.
+"""
+abstract type AbstractSoilCO2BC{FT} <: ClimaLSM.AbstractBC{FT} end
+
+
+struct SoilCO2FluxBC{FT} <: AbstractSoilCO2BC{FT}
+    bc::Function
+end
+
+"""
+ClimaLSM.boundary_flux
+
+test
+"""
+function ClimaLSM.boundary_flux(
+    bc::SoilCO2FluxBC{FT},
+    boundary::ClimaLSM.AbstractBoundary,
+    Δz::ClimaCore.Fields.Field,
+    Y::ClimaCore.Fields.FieldVector,
+    p::ClimaCore.Fields.FieldVector,
+    t::FT,
+)::ClimaCore.Fields.Field where {FT}
+    return bc.bc(p, t) .+ ClimaCore.Fields.zeros(axes(Δz))
+end
+
+struct SoilCO2StateBC{FT} <: AbstractSoilCO2BC{FT}
+    bc::Function
+end
+
+function ClimaLSM.boundary_flux(
+    bc::SoilCO2StateBC,
+    ::ClimaLSM.TopBoundary,
+    Δz,
+    Y,
+    p,
+    t,
+)::ClimaCore.Fields.Field
+    p_len = Spaces.nlevels(axes(p.DETECT.D))
+    D_c = Fields.level(p.DETECT.D, p_len)
+    C_c = Fields.level(Y.DETECT.C, p_len)
+    C_bc = bc.bc(p, t)
+    return ClimaLSM.diffusive_flux(D_c, C_bc, C_c, Δz)
+end
+
+function ClimaLSM.boundary_flux(
+    bc::SoilCO2StateBC,
+    ::ClimaLSM.BottomBoundary,
+    Δz,
+    Y,
+    p,
+    t,
+)::ClimaCore.Fields.Field
+    D_c = Fields.level(p.DETECT.D, 1)
+    C_c = Fields.level(Y.DETECT.C, 1)
+    C_bc = bc.bc(p, t)
+    return ClimaLSM.diffusive_flux(D_c, C_c, C_bc, Δz)
+end
+
+include("./co2_parameterizations.jl") # is this needed?
```

Comment:
I believe so. I think we want those functions to be part of this module

Reply:
Right! of course. 
I noticed we included that file at the top of the module previously, but I imagine it doesn't matter